pyrubberband/pyrb.py

```diff
@@ -141,6 +141,66 @@ def time_stretch(y, sr, rate, rbargs=None):
     return __rubberband(y, sr, **rbargs)
 
 
+def timemap_stretch(y, sr, time_map, rbargs=None):
+    '''Apply a timemap stretch to an audio time series.
+
+    This uses the `time` and `timemap` form for rubberband.
+
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,) or (n, c)]
+        Audio time series, either single or multichannel
+
+    sr : int > 0
+        Sampling rate of `y`
+
+    time_map : list
+        Each element is a tuple `t` of length 2 which correspond to the input
+         frame and desired output frame.
+        If t[1] < t[0] the track will be sped up in this area.
+        time_map[-1] must correspond to the lengths of the input audio and
+         output audio.
+
+    rbargs
+        Additional keyword parameters for rubberband
+
+        See `rubberband -h` for details.
+
+    Returns
+    -------
+    y_stretch : np.ndarray
+        Time-stretched audio
+
+    Raises
+    ------
+    ValueError
+        if `rate <= 0`
```

Comment:
there is no `rate` parameter in this function, but there should be a value check that the input mapping is coherent (ie monotonic)